packages/chat-stateful-client/src/StatefulChatThreadClient.ts

```diff
@@ -120,11 +120,13 @@ class ProxyChatThreadClient implements ProxyHandler<ChatThreadClient> {
       }
       case 'updateTopic': {
         return async (...args: Parameters<ChatThreadClient['updateTopic']>) => {
-          const result = await chatThreadClient.updateTopic(...args);
-          const [topic] = args;
+          return this._context.asyncTeeErrorToState(async () => {
```

Comment:
I guess we might need to wrap every function with this in the future? Or is it just for this one?

Since we are already using proxy, maybe we could just add a single wrapper for all the functions using proxy.